modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/TableColumnHeaderTest.java

```diff
@@ -0,0 +1,211 @@
+
+        sl = new StageLoader(tableView);
+        Toolkit tk = Toolkit.getToolkit();
+
+        tk.firePulse();
+        //Force the column to have default font, otherwise font Amble is applied and mess with header width size
+        column.setStyle("-fx-font: System;");
+        firstColumnHeader = VirtualFlowTestUtils.getTableColumnHeader(tableView, column);
+    }
+
+    @After
+    public void after(){
+        sl.dispose();
+    }
+
+    /**
+     * @test
+     * @bug 8207957
+     * Resize the column header without modifications
+     */
+    @Test
+    public void test_resizeColumnToFitContent() {
+        TableColumn column = tableView.getColumns().get(0);
+        double width = column.getWidth();
+        TableColumnHeaderShim.resizeColumnToFitContent(firstColumnHeader, -1);
+
+        assertEquals("Width must be the same",
+                width, column.getWidth(), 0.001);
+    }
+
+    /**
+     * @test
+     * @bug 8207957
+     * Resize the column header with first column increase
+     */
+    @Test
+    public void test_resizeColumnToFitContentIncrease() {
+        TableColumn column = tableView.getColumns().get(0);
+        double width = column.getWidth();
+
+        tableView.getItems().get(0).setFirstName("This is a big text inside that column");
+
+        assertEquals("Width must be the same",
+                width, column.getWidth(), 0.001);
+
+        TableColumnHeaderShim.resizeColumnToFitContent(firstColumnHeader, -1);
+        assertTrue("Column width must be greater",
+                width < column.getWidth());
+
+        //Back to initial value
+        tableView.getItems().get(0).setFirstName("Humphrey McPhee");
+
+        TableColumnHeaderShim.resizeColumnToFitContent(firstColumnHeader, -1);
+        assertEquals("Width must be equal to initial value",
+                width, column.getWidth(), 0.001);
+    }
+
+    /**
+     * @test
+     * @bug 8207957
+     * Resize the column header with first column decrease
+     */
+    @Test
+    public void test_resizeColumnToFitContentDecrease() {
+        TableColumn column = tableView.getColumns().get(0);
+        double width = column.getWidth();
+
+        tableView.getItems().get(0).setFirstName("small");
+        tableView.getItems().get(1).setFirstName("small");
+        tableView.getItems().get(2).setFirstName("small");
+        tableView.getItems().get(3).setFirstName("small");
+
+        assertEquals("Width must be the same",
+                width, column.getWidth(), 0.001);
+
+        TableColumnHeaderShim.resizeColumnToFitContent(firstColumnHeader, -1);
+        assertTrue("Column width must be smaller",
+                width > column.getWidth());
+
+        //Back to initial value
+        tableView.getItems().get(0).setFirstName("Humphrey McPhee");
+        tableView.getItems().get(1).setFirstName("Justice Caldwell");
+        tableView.getItems().get(2).setFirstName("Orrin Davies");
+        tableView.getItems().get(3).setFirstName("Emma Wilson");
```

Comment:
It would be nice to not have to repeat the set of strings used. I recommend putting them in a static string array (they appears either 2 or 3 times in the test file). Maybe create 4 static strings for the 4 names?